lms/djangoapps/course_api/views.py

```diff
@@ -253,3 +256,100 @@ def get_queryset(self):
             filter_=form.cleaned_data['filter_'],
             search_term=form.cleaned_data['search_term']
         )
+
+
+class CourseIdListUserThrottle(UserRateThrottle):
+    """Limit the number of requests users can make to the course list id API."""
+
+    THROTTLE_RATES = {
+        'user': '20/minute',
+        'staff': '40/minute',
+    }
+
+    def allow_request(self, request, view):
+        # Use a special scope for staff to allow for a separate throttle rate
+        user = request.user
+        if user.is_authenticated and (user.is_staff or user.is_superuser):
+            self.scope = 'staff'
+            self.rate = self.get_rate()
+            self.num_requests, self.duration = self.parse_rate(self.rate)
+
+        return super(CourseIdListUserThrottle, self).allow_request(request, view)
+
+
+@view_auth_classes()
+class CourseIdListView(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+
+        Request a list of course IDs for all courses the specified user can
+        access based on the provided parameters.
+
+    **Example Requests**
+
+        GET /api/courses/v1/courses_ids/
+
+    **Response Values**
+
+        Body comprises a list of course ids and pagination details.
+
+    **Parameters**
+
+        username (optional):
+            The username of the specified user whose visible courses we
+            want to see.
+
+        role (required):
+            Course ids are filtered such that only those for which the
+            user has the specified role are returned. Role can be "staff"
+            or "instructor".
+            Case-insensitive.
+
+    **Returns**
+
+        * 200 on success, with a list of course ids and pagination details
+        * 400 if an invalid parameter was sent or the username was not provided
+          for an authenticated request.
+        * 403 if a user who does not have permission to masquerade as
+          another user who specifies a username other than their own.
+        * 404 if the specified user does not exist, or the requesting user does
+          not have permission to view their courses.
+
+        Example response:
+
+            {
+                "count": 2,
+                "results": [
+                    "course-v1:edX+DemoX+Demo_Course"
+                ],
+                "current_page": 1,
+                "start": 0,
+                "next": null,
+                "previous": null,
+                "num_pages": 1
+            }
```

Comment:
**Reviewer Notes:**
1. Count is returning the wrong value. I'm not sure if I am doing something wrong, or if `LazySequence` is updating the accurate count fast enough for the pagination and if this is a potentially wider spread issue.
2. I used `DefaultPagination`, which seems to be the best practice, although we have multiple classes in `edx-drf-extensions`.  Below is the output using `NamespacedPageNumberPagination`, as used in the courses api.  The API Conventions doc makes it seem like `DefaultPagination` is the recommendation, although I like `NamespacedPageNumberPagination` better.  I like clear answers. :(
```
            {
                "pagination": {
                    "num_pages": 1,
                    "count": 2,
                    "next": null,
                    "previous": null
                },
                "results": [
                    "course-v1:edX+DemoX+Demo_Course",
                ]
            }
```

Reply:
@feanil: 
1. I decided to switch to `NamespacedPageNumberPagination` to be locally consistent (and I prefer it).
2. I will look into the bad count.